net-misc/networkmanager: Version bump to 1.32.12

jsmolic opened the pull request:
- Add USE=docs (enabled by default) and require it if USE=introspection,
  since building docs requires introspection support. This should also
  allow us to fix the build issue when building without introspection by
  turning off doc generation as well.
- Sort dependencies alphabetically and remove outdated version
  specifiers
- Couple of minor syntax changes (adding a missing `|| die`, changing
  `rm -rf` to `rm -r`, etc.)

Bug: https://bugs.gentoo.org/811564
Signed-off-by: Jakov Smolić <jsmolic@gentoo.org>

```diff
--- networkmanager-1.32.10.ebuild
+++ networkmanager-1.32.12.ebuild
@@ -14,12 +14,13 @@
 LICENSE="GPL-2+ LGPL-2.1+"
 SLOT="0"

-IUSE="audit bluetooth +concheck connection-sharing debug dhclient dhcpcd elogind gnutls +introspection iptables iwd kernel_linux psl lto +nss nftables +modemmanager ofono ovs policykit +ppp resolvconf selinux syslog systemd teamd test +tools vala +wext +wifi"
+IUSE="audit bluetooth +concheck connection-sharing debug dhclient dhcpcd +doc elogind gnutls +introspection iptables iwd kernel_linux psl lto +nss nftables +modemmanager ofono ovs policykit +ppp resolvconf selinux syslog systemd teamd test +tools vala +wext +wifi"
 RESTRICT="!test? ( test )"

 REQUIRED_USE="
 	bluetooth? ( modemmanager )
 	connection-sharing? ( || ( iptables nftables ) )
+	doc? ( introspection )
 	iwd? ( wifi )
 	vala? ( introspection )
 	wext? ( wifi )
@@ -32,83 +33,83 @@
 KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~riscv ~sparc ~x86"

 COMMON_DEPEND="
-	sys-apps/util-linux[${MULTILIB_USEDEP}]
-	elogind? ( >=sys-auth/elogind-219 )
-	>=virtual/libudev-175:=[${MULTILIB_USEDEP}]
-	sys-apps/dbus
+	dev-libs/glib:2[${MULTILIB_USEDEP}]
 	net-libs/libndp
-	systemd? ( >=sys-apps/systemd-209:0= )
-	>=dev-libs/glib-2.40:2[${MULTILIB_USEDEP}]
-	introspection? ( >=dev-libs/gobject-introspection-0.10.3:= )
-	selinux? ( sys-libs/libselinux )
+	sys-apps/dbus
+	sys-apps/util-linux[${MULTILIB_USEDEP}]
+	virtual/libudev:=[${MULTILIB_USEDEP}]
 	audit? ( sys-process/audit )
-	teamd? (
-		>=dev-libs/jansson-2.7
-		>=net-misc/libteam-1.9
+	bluetooth? ( net-wireless/bluez )
+	concheck? ( net-misc/curl )
+	connection-sharing? (
+		net-dns/dnsmasq[dbus,dhcp]
+		iptables? ( net-firewall/iptables )
+		nftables? ( net-firewall/nftables )
 	)
-	policykit? ( >=sys-auth/polkit-0.106 )
-	nss? ( >=dev-libs/nss-3.11:=[${MULTILIB_USEDEP}] )
+	dhclient? ( net-misc/dhcp[client] )
+	dhcpcd? ( net-misc/dhcpcd )
+	elogind? ( sys-auth/elogind )
 	gnutls? (
-		>=net-libs/gnutls-2.12:=[${MULTILIB_USEDEP}]
+		net-libs/gnutls:=[${MULTILIB_USEDEP}]
 	)
-	ppp? ( >=net-dialup/ppp-2.4.5:=[ipv6] )
+	introspection? ( dev-libs/gobject-introspection:= )
 	modemmanager? (
 		net-misc/mobile-broadband-provider-info
-		>=net-misc/modemmanager-0.7.991:0=
+		net-misc/modemmanager:0=
 	)
-	bluetooth? ( >=net-wireless/bluez-5 )
+	nss? ( dev-libs/nss:=[${MULTILIB_USEDEP}] )
 	ofono? ( net-misc/ofono )
-	dhclient? ( >=net-misc/dhcp-4[client] )
-	dhcpcd? ( >=net-misc/dhcpcd-9.3.3 )
-	ovs? ( >=dev-libs/jansson-2.7 )
+	ovs? ( dev-libs/jansson )
+	policykit? ( sys-auth/polkit )
+	ppp? ( net-dialup/ppp:=[ipv6] )
+	psl? ( net-libs/libpsl )
 	resolvconf? ( net-dns/openresolv )
-	connection-sharing? (
-		net-dns/dnsmasq[dbus,dhcp]
-		iptables? ( net-firewall/iptables )
-		nftables? ( net-firewall/nftables )
+	selinux? ( sys-libs/libselinux )
+	systemd? ( sys-apps/systemd:0= )
+	teamd? (
+		dev-libs/jansson
+		net-misc/libteam
 	)
-	psl? ( net-libs/libpsl )
-	concheck? ( net-misc/curl )
 	tools? (
 		sys-libs/readline:0=
-		>=dev-libs/newt-0.52.15
+		dev-libs/newt
 	)
 "
 RDEPEND="${COMMON_DEPEND}
 	acct-group/plugdev
 	|| (
-		net-misc/iputils[arping(+)]
 		net-analyzer/arping
+		net-misc/iputils[arping(+)]
 	)
 	wifi? (
-		!iwd? ( >=net-wireless/wpa_supplicant-0.7.3-r3[dbus] )
+		!iwd? ( net-wireless/wpa_supplicant[dbus] )
 		iwd? ( net-wireless/iwd )
 	)
 "
 DEPEND="${COMMON_DEPEND}
-	>=sys-kernel/linux-headers-3.18
 	net-libs/libndp[${MULTILIB_USEDEP}]
+	>=sys-kernel/linux-headers-3.18
 "
 BDEPEND="
+	app-text/docbook-xml-dtd:4.1.2
 	dev-util/gdbus-codegen
 	dev-util/glib-utils
 	dev-util/gtk-doc
-	app-text/docbook-xml-dtd:4.1.2
-	>=dev-util/intltool-0.40
-	>=sys-devel/gettext-0.17
+	dev-util/intltool
+	sys-devel/gettext
 	virtual/pkgconfig
 	introspection? (
 		$(python_gen_any_dep 'dev-python/pygobject:3[${PYTHON_USEDEP}]')
 		dev-lang/perl
 		dev-libs/libxslt
 	)
-	vala? ( $(vala_depend) )
 	test? (
-		>=dev-libs/jansson-2.7
+		dev-libs/jansson
 		$(python_gen_any_dep '
 			dev-python/dbus-python[${PYTHON_USEDEP}]
 			dev-python/pygobject:3[${PYTHON_USEDEP}]')
 	)
+	vala? ( $(vala_depend) )
 "

 python_check_deps() {
@@ -155,6 +156,7 @@
 		fi
 		linux-info_pkg_setup
 	fi
+
 	if use introspection || use test; then
 		python-any-r1_pkg_setup
 	fi
@@ -227,7 +229,7 @@

 		$(meson_native_use_bool introspection)
 		$(meson_native_use_bool vala vapi)
-		$(meson_native_true docs)
+		$(meson_native_use_bool doc docs)
 		-Dtests=$(multilib_native_usex test)
 		$(meson_native_true firewalld_zone)
 		-Dmore_asserts=0
@@ -298,7 +300,7 @@
 multilib_src_install() {
 	meson_src_install
 	if ! multilib_is_native_abi; then
-		rm -rf "${ED}"/{etc,usr/{bin,lib/NetworkManager,share},var}
+		rm -r "${ED}"/{etc,usr/{bin,lib/NetworkManager,share},var} || die
 	fi
 }

@@ -322,14 +324,14 @@

 	# Allow users in plugdev group to modify system connections
 	insinto /usr/share/polkit-1/rules.d/
-	doins "${FILESDIR}/01-org.freedesktop.NetworkManager.settings.modify.system.rules"
+	doins "${FILESDIR}"/01-org.freedesktop.NetworkManager.settings.modify.system.rules

 	insinto /usr/lib/NetworkManager/conf.d #702476
 	doins "${S}"/examples/nm-conf.d/31-mac-addr-change.conf

 	if use iwd; then
 		# This goes to $nmlibdir/conf.d/ and $nmlibdir is '${prefix}'/lib/$PACKAGE, thus always lib, not get_libdir
-		cat <<-EOF > "${ED}"/usr/lib/NetworkManager/conf.d/iwd.conf
+		cat <<-EOF > "${ED}"/usr/lib/NetworkManager/conf.d/iwd.conf || die
 		[device]
 		wifi.backend=iwd
 		EOF
```

telans commented:
As a request would you please mind adding a `tc-is-gcc` check for `USE=lto`? https://bugs.gentoo.org/809695

I was about to create a PR regarding this but it might be quicker for this to be in here too? Thanks

jsmolic commented:
> As a request would you please mind adding a `tc-is-gcc` check for `USE=lto`? https://bugs.gentoo.org/809695
> 
> I was about to create a PR regarding this but it might be quicker for this to be in here too? Thanks, no worries if not.

Done!

telans commented:
> > As a request would you please mind adding a `tc-is-gcc` check for `USE=lto`? https://bugs.gentoo.org/809695
> > I was about to create a PR regarding this but it might be quicker for this to be in here too? Thanks, no worries if not.
> 
> Done!

Thank you!

thesamesam commented:
There's some GNOME rule about sorting deps but I have no idea what it is. At a skim, it's not listed [here](https://wiki.gentoo.org/wiki/Project:GNOME/Gnome_Team_Ebuild_Policies). But I'd therefore keep the order as is until you get clarification.

leio commented:
Once updated for feedback, I'd appreciate a diff from previous version again for a quick initial review

jsmolic commented:
> Do not arbitrarily sort alphabetically indeed. It gives no value, especially when it's just sorted by category name. In GNOME we sort by the order of occurrence in the build system, because then it's easy to skim things from top to bottom, side by side and see everything is still up to date. Relatedly, we order the emesonargs stuff in the same order as they appear in upstream meson_options.txt.

Okay, I respect gnome team's styling as I know it can differ from project to project, for me this part is not that important as it's pure syntax so I probably shouldn't have changed it in the first place. I've restored the original dependencies order.

> API docs in the format that devhelp can show must be behind IUSE=gtk-doc, if optional. Man pages must always be installed if it doesn't mean heavy deps. It looks like they might be under same meson option and need some patching and work with upstream perhaps?
> 
Hmm, okay, I can change it to `gtk-doc` but from what I've looked at the build system it doesn't support building docs without introspection support, as for man pages it seems they are not build if USE=-doc, it could be brought up upstream but I think this could still be useful change as we can't currently build networkmanager without introspection.

> Also inherit order very much matters and sorting them alphabetically can have side-effects. I typically order the stuff that matters (things I need the exported phases from) to the end to ensure those are what get used. The changes here about that look OK though for inherits, as there's a src_prepare override anyways and meson_src_prepare and vala_src_prepare won't clash.
> 
Sure, I know that order inherit is important, though in this case it does not change anything as no exported function is overwritten, I've added toolchain-funcs.eclass to `tc-is-gcc` check, I didn't know where to add it so I sorted inherits alphabetically.

> Also you are arbitrarily killing minimum version requirements - don't do that, they are there based on build system or other needs. There being no older version available doesn't mean it should stop having the correct minimum dependency based on build system or otherwise. We track that, which is easy thanks to the ordering of deps in meson.build occurrence order, and it helps for users that haven't updated the machine for a long time too.
> 
Reverted to how it is usually with version specifiers.
> I can't review the changeset for the actual changes with so many dependency changes going on.

I'm doing some minor changes, I'll paste the diff again after I finish.

jsmolic commented:
```diff
--- networkmanager-1.32.10-r1.ebuild
+++ networkmanager-1.32.12.ebuild
@@ -6,7 +6,7 @@
 VALA_USE_DEPEND="vapigen"
 PYTHON_COMPAT=( python3_{7..10} )

-inherit gnome.org linux-info meson-multilib python-any-r1 systemd readme.gentoo-r1 vala virtualx udev
+inherit gnome.org linux-info meson-multilib python-any-r1 readme.gentoo-r1 systemd toolchain-funcs udev vala virtualx

 DESCRIPTION="A set of co-operative tools that make networking simple and straightforward"
 HOMEPAGE="https://wiki.gnome.org/Projects/NetworkManager"
@@ -14,12 +14,13 @@
 LICENSE="GPL-2+ LGPL-2.1+"
 SLOT="0"

-IUSE="audit bluetooth +concheck connection-sharing debug dhclient dhcpcd elogind gnutls +introspection iptables iwd kernel_linux psl lto +nss nftables +modemmanager ofono ovs policykit +ppp resolvconf selinux syslog systemd teamd test +tools vala +wext +wifi"
+IUSE="audit bluetooth +concheck connection-sharing debug dhclient dhcpcd elogind gnutls +gtk-doc +introspection iptables iwd kernel_linux psl lto +nss nftables +modemmanager ofono ovs policykit +ppp resolvconf selinux syslog systemd teamd test +tools vala +wext +wifi"
 RESTRICT="!test? ( test )"

 REQUIRED_USE="
 	bluetooth? ( modemmanager )
 	connection-sharing? ( || ( iptables nftables ) )
+	gtk-doc? ( introspection )
 	iwd? ( wifi )
 	vala? ( introspection )
 	wext? ( wifi )
@@ -29,7 +30,7 @@
 	?? ( syslog systemd )
 "

-KEYWORDS="~alpha amd64 ~arm arm64 ~ia64 ~ppc ~ppc64 ~riscv ~sparc ~x86"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~riscv ~sparc ~x86"

 COMMON_DEPEND="
 	sys-apps/util-linux[${MULTILIB_USEDEP}]
@@ -155,9 +156,16 @@
 		fi
 		linux-info_pkg_setup
 	fi
+
 	if use introspection || use test; then
 		python-any-r1_pkg_setup
 	fi
+
+	# bug 809695
+	if tc-is-clang && use lto; then
+		eerror "Clang does not support -flto-partition"
+		die "Please use gcc or turn off USE=lto flag when building with clang"
+	fi
 }

 src_prepare() {
@@ -227,7 +235,7 @@

 		$(meson_native_use_bool introspection)
 		$(meson_native_use_bool vala vapi)
-		$(meson_native_true docs)
+		$(meson_native_use_bool gtk-doc docs )
 		-Dtests=$(multilib_native_usex test)
 		$(meson_native_true firewalld_zone)
 		-Dmore_asserts=0
@@ -298,7 +306,7 @@
 multilib_src_install() {
 	meson_src_install
 	if ! multilib_is_native_abi; then
-		rm -rf "${ED}"/{etc,usr/{bin,lib/NetworkManager,share},var}
+		rm -r "${ED}"/{etc,usr/{bin,lib/NetworkManager,share},var} || die
 	fi
 }

@@ -322,14 +330,14 @@

 	# Allow users in plugdev group to modify system connections
 	insinto /usr/share/polkit-1/rules.d/
-	doins "${FILESDIR}/01-org.freedesktop.NetworkManager.settings.modify.system.rules"
+	doins "${FILESDIR}"/01-org.freedesktop.NetworkManager.settings.modify.system.rules

 	insinto /usr/lib/NetworkManager/conf.d #702476
 	doins "${S}"/examples/nm-conf.d/31-mac-addr-change.conf

 	if use iwd; then
 		# This goes to $nmlibdir/conf.d/ and $nmlibdir is '${prefix}'/lib/$PACKAGE, thus always lib, not get_libdir
-		cat <<-EOF > "${ED}"/usr/lib/NetworkManager/conf.d/iwd.conf
+		cat <<-EOF > "${ED}"/usr/lib/NetworkManager/conf.d/iwd.conf || die
 		[device]
 		wifi.backend=iwd
 		EOF
```

thesamesam commented:
I'll merge this given the build failure bug has racked up CCs. Changes seem rather small now compared to the last version and not aware of any contradictions with GNOME policy now that the dependency ordering has been sorted.
